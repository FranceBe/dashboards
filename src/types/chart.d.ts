export type ChartProps = {
  data: Record<string, string | number>[]
  valueKey: string
  nameKey: string
}
