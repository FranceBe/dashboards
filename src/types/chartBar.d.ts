import any = jasmine.any

export type ChartBarProps = {
  data: Record<string, string | number>[]
  valueKey: string
  nameKey: string
}
