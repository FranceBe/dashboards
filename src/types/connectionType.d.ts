export type ConnectionTypeProps = {
  connection_type: 'ethernet' | 'wifi' | 'cellular'
}
