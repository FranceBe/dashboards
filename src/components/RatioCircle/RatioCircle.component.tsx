// Component that display a circle that can display content and
// Can accept a specific color as border-color
import { RatioContainer } from 'components/RatioCircle/ratioCircle.style'
import React from 'react'
import { RatioCircleProps } from 'types/ratioCircle'
export const RatioCircle: React.FC<RatioCircleProps> = ({ content, color }) => {
  return (
    <RatioContainer data-testid={'ratio-circle'} color={color}>
      {content}
    </RatioContainer>
  )
}

export default RatioCircle
