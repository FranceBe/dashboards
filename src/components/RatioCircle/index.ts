import RatioCircleComponent, { RatioCircle } from 'components/RatioCircle/RatioCircle.component'

export { RatioCircle }

export default RatioCircleComponent
