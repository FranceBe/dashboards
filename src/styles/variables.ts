// Colors to use matching the graphic chart
export const palette: { [key: string]: string } = {
  blue_light: '#f0f0fc',
  green_primary: '#65db25',
  grey_primary: '#343536',
  red_primary: '#eb4034',
}

// Spaces to use to respect an 8px grid
export const spaces: { [key: string]: string } = {
  large: '24px',
  medium: '16px',
  regular: '8px',
  small: '4px',
}

// Font sizes to use in project
export const fontSizes: { [key: string]: string } = {
  small: '12px',
}
