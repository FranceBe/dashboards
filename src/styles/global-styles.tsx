import { createGlobalStyle } from 'styled-components'
import { palette } from 'styles/variables'

export const GlobalStyle = createGlobalStyle`
  body {
    margin: 1.5em;
    background-color: ${palette.blue_light};
    color: ${palette.grey_primary};
  }
  a {
    text-decoration: none;
    :active, :visited {
      color: ${palette.blue_primary};
    }
    :hover, :visited:hover {
      color: ${palette.blue_secondary};
    }
  }
`
