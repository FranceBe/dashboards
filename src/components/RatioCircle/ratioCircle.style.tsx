import React from 'react'
import styled from 'styled-components'
import { elements, fontSizes, palette } from 'styles/variables'

export const RatioContainer = styled((props) => <div {...props} />)<{
  color: string
}>`
  height: 150px;
  width: 150px;
  border: 3px solid ${(props) => props.color || palette.grey_primary};
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  place-self: center;
  box-shadow: ${elements.box_shadow};
  font-size: ${fontSizes.large};
`
