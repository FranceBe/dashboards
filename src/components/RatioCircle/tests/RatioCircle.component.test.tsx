import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { render, screen } from '@testing-library/react'
import RatioCircle from 'components/RatioCircle'
import React from 'react'
import { palette } from 'styles/variables'
import { RatioCircleProps } from 'types/ratioCircle'

describe('RatioCircle', () => {
  const defaultProps: RatioCircleProps = {
    content: '14/23',
  }
  it('should match snapshot', () => {
    const { container } = render(<RatioCircle {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should have grey_primary border color by default', () => {
    const { container } = render(<RatioCircle {...defaultProps} />)
    expect(container.firstChild).toHaveStyleRule('border', `3px solid ${palette.grey_primary}`)
  })
  it('should have custom border color when color props is provided', () => {
    const { container } = render(<RatioCircle {...defaultProps} color={'#003725'} />)
    expect(container.firstChild).toHaveStyleRule('border', '3px solid #003725')
  })
  it('should display content', () => {
    render(<RatioCircle {...defaultProps} color={'#003725'} />)
    expect(screen.getByText('14/23')).toBeInTheDocument()
  })
})
