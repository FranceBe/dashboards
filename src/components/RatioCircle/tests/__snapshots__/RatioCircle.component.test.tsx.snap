// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatioCircle should match snapshot 1`] = `
.c0 {
  height: 150px;
  width: 150px;
  border: 3px solid #343536;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  place-self: center;
  box-shadow: 0 5px 25px 0 rgba(0,0,45,0.1),0 2px 6px 0 rgba(0,0,0,0.04), 0 16px 24px 0 rgba(0,0,46,0.06);
  font-size: 1.5rem;
}

<div
  class="c0"
  data-testid="ratio-circle"
>
  14/23
</div>
`;
